src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -629,18 +713,43 @@ private static BlittableJsonReaderObject CreateHalfDocument(DocumentsOperationCo
                         builder.WriteValue(lsv);
                     else
                         throw new InvalidDataException("Unknown type: " + prop.Token + " when trying to split counter doc");
+
+                    propNames.Add(prop.Name);
                 }
 
                 builder.WriteObjectEnd();
 
+                builder.WritePropertyName(CounterNames);
```

Comment:
Why is this an array and not a dictionary? That would allow O(1) searches, no?

Reply:
changed from BlittableArray to BlittableObject